controllers/healthcheck_controller.go

```diff
@@ -138,25 +139,38 @@ func (r *HealthCheckReconciler) processHealthCheck(ctx context.Context, req ctrl
 	if hcSpec.Workflow.Resource != nil {
 		wfNamePrefix := hcSpec.Workflow.GenerateName
 		wfNamespace := hcSpec.Workflow.Resource.Namespace
-		repeatAfterSec := hcSpec.RepeatAfterSec
 		level := hcSpec.Level
 		sa := hcSpec.Workflow.Resource.ServiceAccount
 		now := metav1.Time{Time: time.Now()}
 		var finishedAtTime int64
 		if healthCheck.Status.FinishedAt != nil {
 			finishedAtTime = healthCheck.Status.FinishedAt.Time.Unix()
 		}
-		// workflows can be paused by setting repeatAfterSec to <= 0.
-		if repeatAfterSec <= 0 {
-			log.Info("Workflow will be skipped due to repeatAfterSec value", "repeatAfterSec", repeatAfterSec)
+
+		// workflows can be paused by setting repeatAfterSec to <= 0 and not specifying the schedule for cron.
+		if hcSpec.RepeatAfterSec <= 0 && hcSpec.Scheduler.Cron == "" {
+			log.Info("Workflow will be skipped due to repeatAfterSec value", "repeatAfterSec", hcSpec.RepeatAfterSec)
 			healthCheck.Status.Status = "Stopped"
-			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped due to spec.repeatAfterSec set to %d", repeatAfterSec)
+			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped; either spec.RepeatAfterSec or spec.Scheduler must be provided. spec.RepeatAfterSec set to %d. spec.Scheduler set to %v+", hcSpec.RepeatAfterSec, hcSpec.Scheduler)
 			healthCheck.Status.FinishedAt = &now
 			return ctrl.Result{}, nil
-		} else if int(time.Now().Unix()-finishedAtTime) < repeatAfterSec {
+		} else if hcSpec.RepeatAfterSec <= 0 && hcSpec.Scheduler.Cron != "" {
+			log.Info("Workflow to be set with Scheduler", "Cron", hcSpec.Scheduler.Cron)
+			scheduler, err := cron.ParseStandard(hcSpec.Scheduler.Cron)
+			if err != nil {
+				log.Error(err, "fail to parse cron")
+			}
+			// The value from scheduler next and substracting from current time is in fraction as we convert to int it will be 1 less than
+			// the intended reschedule so we need to add 1sec to get the actual value
+			// we need to update the spec so have to healthCheck.Spec.RepeatAfterSec instead of local variable hcSpec
+			healthCheck.Spec.RepeatAfterSec = int(scheduler.Next(time.Now()).Sub(time.Now())/time.Second) + 1
+			// log.Info("spec.RepeatAfterSec value is set", "RepeatAfterSec", healthCheck.Spec.RepeatAfterSec)
```

Comment:
looks like this commented out log line is same as un-commented one below... could probably remove this line
```suggestion
```